gcpvault.go

```diff
@@ -77,39 +77,88 @@ func GetSecrets(ctx context.Context, cfg Config) (map[string]interface{}, error)
 		return getLocalSecrets(ctx, cfg)
 	}
 
-	// create signed JWT with our service account
-	jwt, err := newJWT(ctx, cfg)
+	vClient, err := login(ctx, cfg)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to create JWT")
+		return nil, errors.Wrap(err, "unable to login to vault")
 	}
 
-	// init vault client
-	vcfg := api.DefaultConfig()
-	vcfg.MaxRetries = cfg.MaxRetries
-	vcfg.Address = cfg.VaultAddress
-	vcfg.HttpClient = getHTTPClient(ctx)
-	vClient, err := api.NewClient(vcfg)
+	// fetch secrets
+	secrets, err := vClient.Logical().Read(cfg.SecretPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get secrets")
+	}
+
+	// if secrets are versioned, we have to go deeper
+	if strings.Contains(cfg.SecretPath, "/versioned/") {
+		return secrets.Data["data"].(map[string]interface{}), nil
+	}
+
+	return secrets.Data, nil
+}
+
+// PutSecrets writes secrets to Vault at the configured path.
+// This is comparable to the `vault write` command.
+func PutSecrets(ctx context.Context, cfg Config, secrets map[string]interface{}) error {
+	vClient, err := login(ctx, cfg)
+	if err != nil {
+		return errors.Wrap(err, "unable to login to vault")
+	}
+
+	_, err = vClient.Logical().Write(cfg.SecretPath, secrets)
+	return errors.Wrap(err, "unable to make vault request")
+}
+
+// PutVersionedSecrets writes versioned secrets to Vault at the configured path.
+// The path should be in the form `org/repo/versioned/data/my-secret`
```

Comment:
From looking at the Vault docs, this path convention is NYT specific.

Can we add a new config option for the "versioned" substring to look for, but have it default to `/versioned/` via the envconfig tag?